### PR TITLE
refactor: move log_block class from mutation_log.h to separated file

### PR DESCRIPTION
**Pull request by neverchanje**
mutation_log.h is a long file with many class definitions. This PR simply moves some classes, including `log_block_header` and `log_block`, into a separate file, namely 'log_block.h'. 

This PR only makes trivial changes.

### CODE REVIEW

File: src/dist/replication/lib/log_block.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+namespace dsn {
+namespace replication {
+
+// each block in log file has a log_block_header
+struct log_block_header
+{
+    int32_t magic;    // 0xdeadbeef
+    int32_t length;   // block data length (not including log_block_header)
+    int32_t body_crc; // block data crc (not including log_block_header)
+    uint32_t
```
Comment (levy5307) [inline src/dist/replication/lib/log_block.h]: remote the line break

File: src/dist/replication/lib/mutation.cpp [hunk 2]
```diff
@@ -164,7 +164,7 @@ void mutation::add_client_request(task_code code, dsn::message_ex *request)
     dassert(client_requests.size() == data.updates.size(), "size must be equal");
 }
 
-void mutation::write_to(std::function<void(const blob &)> inserter) const
+void mutation::write_to(const std::function<void(const blob &)> &inserter) const
```
Comment (levy5307) [inline src/dist/replication/lib/mutation.cpp]: Why are you delete the `const` and `&` ?

File: src/dist/replication/lib/mutation_log.cpp [hunk 3]
```diff
@@ -797,7 +797,7 @@ error_code mutation_log::create_new_log_file()
         header_len = logf->write_file_header(temp_writer, _shared_log_info_map);
     }
 
-    log_block *blk = logf->prepare_log_block();
```
Comment (levy5307) [inline src/dist/replication/lib/mutation_log.cpp]: There will produce memory leak if blk != nullptr.

File: src/dist/replication/lib/log_block.cpp [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace dsn {
+namespace replication {
+
+log_block::log_block() { init(); }
+
+void log_block::init()
+{
+    log_block_header hdr;
+    hdr.magic = 0xdeadbeef;
```
Comment (levy5307) [inline src/dist/replication/lib/log_block.cpp]: I think you should move these code to the constructor of log_block_header

File: src/dist/replication/lib/log_block.cpp [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+// can be found in the LICENSE file in the root directory of this source tree.
+
+#include "log_block.h"
+
+namespace dsn {
+namespace replication {
+
+log_block::log_block() { init(); }
+
+void log_block::init()
```
Comment (hycdong) [inline src/dist/replication/lib/log_block.cpp]: why add a function called `init` here, rather than write those code in construction function?

Comment (neverchanje) [reply src/dist/replication/lib/log_block.cpp]: I will write another constructor (in next PR) that's gonna reuse this function. So I intentionally leave it here.

**Pull request merged**